### ISSUE TITLE
WMS authentication problem

### ISSUE THREAD

**Issue by lionelbargeot** [3liz/lizmap-web-client#671]
Hi
I believe my problem is different from this one : https://github.com/3liz/lizmap-web-client/issues/617
I use QGis server 2.18  and LWC 3.2pre
I use a WMS service with basic authentication in url : 
http://user:password@host/wms
or with using http parameters:
http://host/wms?somewmsparameters&user=user&password=password

With qgis, I can access the service without problem, with configuring authentication with qgis parameters (user and password as http parameters)
With LWC and default configuration, I can access the service without problem.
With LWC and direct external WMS access, my client ask authentication again. So I can see in my debuging console that user and password are not present in the url neither as adress option neither as http parameters.

**Comment by gioman** [comment]
Unsure is related by I noticed recently that QGIS server 2.18 has lost the ability to cascade WMS layers that are behind authentication (tested with UMN Mapserver and basic http authentication).

**Comment by lionelbargeot** [comment]
thanks for your answer gioman.
When the http request is send to qgis server (LWC with default configuration, no direct wms), authentication info are added to request (I can see it in wms server logs) and it works.

**Comment by rldhont** [comment]
@lionelbargeot Do you need that the JS client adds authentication parameter in the case of direct access of external WMS ?

**Comment by lionelbargeot** [comment]
@rldhont , yes I need exactly this. Is there a way to do it with lizmap-plugin config ?

**Comment by rldhont** [comment]
It's an issue to be fixed.

**Comment by rldhont** [comment]
@lionelbargeot I don't want that JS client get the password. It will be better to use lizmap as a proxy without QGIS Server.

**Comment by nboisteault** [comment]
@rldhont I think we should handle additional parameters in URL in general but in this case publishers would have to know they should not use login/password parameters or only in trusted environment.
